src/pal/src/loader/module.cpp

```diff
@@ -1620,6 +1620,8 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
         shortAsciiName = FREEBSD_LIBC;
 #elif defined(__NetBSD__)
         shortAsciiName = "libc.so";
+#elif !HAVE_GNU_LIBNAMES_H
+        shortAsciiName = "libc.musl-x86_64.so";
```

Comment:
I have defined MUSL_LIBC_SO in config when the lib_names header is unavailable and the file exists. If future, when `uClibc` and/or `dietlibc` support will be added or other architectures for these non-glibcs, we can add more clauses in the config section.